protostar/testing/test_suite.py

```diff
@@ -1,20 +1,86 @@
-from dataclasses import dataclass
 from pathlib import Path
-from typing import List, Optional
+from typing import List, Optional, Union
+from typing_extensions import Self
+
+from protostar.cairo.cairo_function_executor import Offset
 
 
-@dataclass(frozen=True)
 class TestCase:
-    test_path: Path
-    test_fn_name: str
-    setup_fn_name: Optional[str] = None
+    def __init__(
+        self, test_path: Path, test_fn_name: str, setup_fn_name: Optional[str] = None
+    ):
+        self.test_path = test_path
+        self.test_fn_name = test_fn_name
+        self.setup_fn_name = setup_fn_name
+
+    def __eq__(self, other: Self) -> bool:
+        return (
+            other.test_path == self.test_path
+            and other.test_fn_name == self.test_fn_name
+            and other.setup_fn_name == self.setup_fn_name
+        )
+
+
+class TestCaseWithOffsets(TestCase):
```

Comment:
Do we need to avoid redundance that much? would't be simpler to just create TestCase/Suite for cairo1 separately? Would be much easier to understand.

Reply:
It's easier to incorporate into current structure this way. Otherwise i'd have to reimplement a big part of test running